addons/mail/models/mail_template.py

```diff
@@ -210,6 +211,16 @@ def generate_recipients(self, results, res_ids):
             records = self.env[self.model].browse(results.keys()).read(['company_id'])
             records_company = {rec['id']: (rec['company_id'][0] if rec['company_id'] else None) for rec in records}
 
+        record_sudo_by_id = None
```

Comment:
but my example should have the same effect no ? even if we are in batch, `_related_res_partner_default_create_values` will never be called if there's no record, and we shouldn't fetch the record fields if we don't try to read them

Reply:
but my example should have the same effect no ? even if we are in batch, _related_res_partner_default_create_values will never be called if there's no record, and we shouldn't fetch the record fields if we don't try to read them

do you have any error / query count issue with my example ?

Reply:
Thanks for insisting, I always forgot that browse just create the resultset without doing query. So your solution is great and simpler. Since python 3.7, order in dict are guarantee as I have read so it should work (3.7 being the minimal version for odoo).